posts/2020-07-30-json-logging-open-liberty-20008.adoc

```diff
@@ -0,0 +1,235 @@
+
+|Remote user according to the WebSphere Application Server specific $WSRU header
+
+|%u
+
+|===
+
+For more information:
+
+* link:{url-prefix}/docs/ref/config/#logging.html[Open Liberty Logging]
+
+* link:{url-prefix}/docs/20.0.0.7/log-trace-configuration.html[Open Liberty Log and Trace Configuration]
+
+
+[#JSONtoSystemOut]
+=== Allow JSON written to System.out to pass without wrapping
+
+In Open Liberty, you have the option to format your server logs in basic or JSON format. When logs are in JSON format, you have to specify the sources (message, trace, accessLog, ffdc, audit) you want to send to `messages.log` or `console.log` and `standard-out`. This feature adds a new option for you to write pre-formatted JSON application logs directly to `System.out` or `System.err`. Previously, when Open Liberty was running with JSON logging enabled, Open Liberty embeds anything written to `System.out` or `System.err` into the message field of a `liberty_message` event. Now, you can write JSON directly to `System.out` or `System.err` without wrapping in the `liberty_message` event.
+
+You can now write your JSON application logs directly to `System.out` or `System.err`, which can then be sent to a log analysis tool, such as the ELK (Elasticsearch, Logstash, Kibana) stack. Previously, when JSON logging was enabled, pre-formatted JSON application logs would look like this:
+
+[source, javascript]
+----
+{
+    "type":"liberty_message",
+    "host":"192.168.0.119",
+    "ibm_userDir":"\/you\/yushan.lin@ibm.com\/Documents\/archived-guide-log4j\/finish\/target\/liberty\/wlp\/usr\
+    ",
+    "ibm_serverName":"log4j.sampleServer",
+    "message":"{\n   \"timeMillis\" : 1587666082123,\n  
+            \"thread\" : \"Default Executor-thread-8\",\n  
+            \"level\" : \"WARN\",\n  
+            \"loggerName\" : \"application.servlet.LibertyServlet\",\n  
+            \"message\" : \"hello liberty servlet warning message!\",\n  
+            \"endOfBatch\" : false,\n  
+            \"loggerFqcn\" : \"org.apache.logging.log4j.spi.AbstractLogger\",\n  
+            \"threadId\" : 53,\n  
+            \"threadPriority\" : 5\n}\r",
+    "ibm_threadId":"00000035",
+    "ibm_datetime":"2020-04-23T14:21:22.124-0400",
+    "module":"SystemOut",
+    "loglevel":"SystemOut",
+    "ibm_methodName":"",
+    "ibm_className":"",
+    "ibm_sequence":"1587666082124_000000000001B",
+    "ext_thread":"Default Executor-thread-8”
+}
+----
+
+
+Now you can output the logs so that pre-formatted JSON logs are not wrapped in `liberty_message` events. Visualization tools such as Kibana can be used to analyze certain fields in your JSON logs. You can analyze both custom-formatted JSON application logs and Liberty JSON logs in the same visualization.
+
+At any given time by setting `appsWriteJson="true"` in the logging element of the `server.xml`, or you can enable this functionality, or you can have it set from the moment the server starts by setting in bootstrap.properties:
```

Comment:
You can enable this functionality by setting appsWriteJson="true" in the logging element of the server.xml, or you ....